### PR TITLE
[LW-9020] Add minimal PostHog tracking to Nami

### PR DESCRIPTION
**Pull request by renanvalentin**

### CODE REVIEW

File: src/ui/app/tabs/trezorTx.jsx [hunk 1]
```diff
@@ -89,11 +92,14 @@ const App = () => {
 
 const root = createRoot(window.document.querySelector(`#${TAB.trezorTx}`));
 root.render(
-  <Main>
-    <Router>
-      <App />
-    </Router>
-  </Main>
+  <AnalyticsProvider view={ExtensionViews.Extended}>
```
Comment (renanvalentin) [inline src/ui/app/tabs/trezorTx.jsx:95]: @DominikGuzei do we want to be more specific? e.g: Trezor

File: src/features/analytics/config.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export const PUBLIC_POSTHOG_HOST = 'https://eu.posthog.com';
+
+export const PRODUCTION_TRACKING_MODE_ENABLED =
+  process.env.PRODUCTION_MODE_TRACKING === 'true';
+
+export const PRODUCTION_NETWORK_ID_TO_POSTHOG_PROJECT_ID_MAP = {
+  ['mainnet']: 6621,
+  ['preprod']: 6620,
+  ['preview']: 6619,
```
Comment (DominikGuzei) [inline src/features/analytics/config.ts:9]: @renanvalentin we will be using the same Posthog project for all networks (that will be different in Nami as compared to Lace), so the network should be an event property instead

File: src/features/analytics/event-tracker.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React, { useEffect } from 'react';
+import debounce from 'debounce';
+import { useCaptureEvent } from './hooks';
+import { Events } from './events';
+
+const PAGE_VIEW_DEBOUNCE_DELAY = 100;
```
Comment (DominikGuzei) [inline src/features/analytics/event-tracker.ts:6]: can you elaborate on why this debounce is necessary?

File: src/features/analytics/hooks.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    consent,
+    async (consent) => {
+      // Allow to set the consent state and store it too
+      await setAnalyticsConsent(consent);
+      setConsentState(consent);
+    },
+  ];
+};
+
+export const useCaptureEvent = () => {
```
Comment (DominikGuzei) [inline src/features/analytics/hooks.ts:43]: beautiful 🥇

File: src/features/analytics/services.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+export const getEventMetadata = async (
+  view: ExtensionViews
+): Promise<PostHogMetadata> => {
+  const [userId, projectId] = await Promise.all([getUserId(), getProjectId()]);
+
+  return {
+    view: view,
+    sent_at_local: new Date().toISOString(),
+    distinct_id: userId,
+    posthog_project_id: projectId,
```
Comment (DominikGuzei) [inline src/features/analytics/services.ts:61]: we need to add the current network as event metadata here

File: src/ui/app/tabs/trezorTx.jsx (hunk 1 above)
Comment (DominikGuzei) [reply src/ui/app/tabs/trezorTx.jsx:95]: For now i think it's fine, and its a decision Danilo and Rhys have to make which properties we want to track

File: src/features/analytics/events.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  ReceiveClick = 'receive | receive | click',
+  ReceiveCopyAddressIconClick = 'receive | receive | copy address icon | click',
+
+  // send
+  SendClick = 'send | send | click',
+  SendTransactionDataReviewTransactionClick = 'send | transaction data | review transaction | click',
+  SendTransactionConfirmationConfirmClick = 'send | transaction confirmation | confirm | click',
+  SendTransactionConfirmed = 'send | transaction confirmed',
+
+  // settings
```
[the remark this answers is not in the document]
Comment (rhyslbw) [reply src/features/analytics/events.ts:26]: MVP scope was defined to limit the effort required, but if it's already done, best to leave.

File: src/features/analytics/events.ts [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+  SettingsNetworkPreviewClick = 'settings | network | preview | click',
+  SettingsNetworkPreprodClick = 'settings | network | preprod | click',
+  SettingsNetworkMainnetClick = 'settings | network | mainnet | click',
+  SettingsNetworkTestnetClick = 'settings | network | testnet | click',
```
Comment (rhyslbw) [inline src/features/analytics/events.ts:30]: Did you end up leaving `testnet` in place @renanvalentin ?

Comment (rhyslbw) [inline src/features/analytics/events.ts:30]: Did you end up leaving testnet in place @renanvalentin ?

File: src/features/analytics/events.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (thedanheller) [reply src/features/analytics/events.ts:26]: LGTM

File: src/features/analytics/events.ts (hunk 7 above)
Comment (renanvalentin) [reply src/features/analytics/events.ts]: Remove it here: [b16443a7930370ba7fe9541766b8255b326c3009](https://github.com/input-output-hk/nami/pull/893/commits/0cca1097adb05f332fa63fd0b6f981319a72d111)

File: src/ui/app/tabs/trezorTx.jsx (hunk 1 above)
Comment (rhyslbw) [reply src/ui/app/tabs/trezorTx.jsx:95]: Let's not bother with this detail, it's not that relevant

**Review by rhyslbw — APPROVED** [review]
This looks good from a code perspective. To confirm this is producing the expected events in PostHog, test engineers will need a build to load and run through the flows.

**Pull request merged**